src/functions-and-arrays.js

```diff
@@ -1,41 +1,100 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
+function maxOfTwoNumbers(a, b) {
+  if (a>b) {
+    return a;
+  } else if (b>a) {
+    return b;
+  } else {
+    return a;
+  }
+}
 
 
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
+function findLongestWord(arr) {
+  if (arr.length === 0) return null;
+  let count = '';
+  for (let value of arr) {
+    if (value.length > count.length) {
+      count = value;
+    }
+  }
+  return count;
+}
 
 
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
+function sumNumbers(numbers) {
+  if (numbers.length === 0) return 0;
+  return numbers.reduce((a, b) => a+b);
+}
 
 
 
 // Iteration #3.1 Bonus:
-function sum() {}
+function sum(arr) {
+  let result = 0;
+  for (let i=0; i<arr.length; i++) {
+    if (typeof(arr[i] === 'number')) {
+      result += arr[i];
+    } else if (typeof(arr[i] === 'string')) {
+      result += arr[i].length;
+    } else if (arr.length === 0) {
+      return 0;
+    } else if (arr.length === 1) {
+      return arr[0];
+    } else if (typeof(arr[i] === 'true')) {
```

Comment:
in this case, typeof would be "boolean" not the true or false keyword. the logic is correct, though :)